tests/error/poly_errors_py312/free_var_in_bound2.err

```diff
@@ -0,0 +1,11 @@
+Error: Free type variable (at $FILE:7:18)
+  | 
+5 | 
+6 | @guppy.struct
+7 | class MyStruct[x: T]:
+  |                   ^ Type variable `T` is unbound
+
+Note: Only struct and function definitions can be generic. Other generic values
+or nested types are not supported.
```

Comment:
he real error here is that `T` isn't `nat` or a mix of type bounds though, right?

Reply:
Hmm I don't want to mention `nat` since that's only a temporary restriction until the monomorphization PR is merged. 

In principle, we should allow every copyable&droppable type here, so this message is coming from trying to parse a type and encountering a type variable that is not declared as a parameter. Not sure if there is an easy way to improve this